app/Models/Foundation/Summit/Summit.php

```diff
@@ -2372,23 +2372,26 @@ public function getSlug($suffix = '-summit')
         return strtolower(preg_replace('/[^a-zA-Z0-9\-]/', '', $res));
     }
 
-
-    public function generateRegistrationSlugPrefix(): void
-    {
-        if (empty($this->registration_slug_prefix)) {
-            $this->registration_slug_prefix = $this->slug;
-        }
-    }
-
     /**
      * @return string
      */
     public function getRegistrationSlugPrefix(): string
     {
-        $this->generateRegistrationSlugPrefix();
         return $this->registration_slug_prefix;
     }
 
+    /**
+     * @param string $registration_slug_prefix
+     * @throws ValidationException
+     */
+    public function setRegistrationSlugPrefix(string $registration_slug_prefix)
+    {
+        // once that a ticket is sold, registration_slug_prefix can't be changed anymore
+        if ($this->getPaidTicketsCount() > 0)
+            throw new ValidationException("Cannot change the registration slug when there are paid tickets.");
+        $this->registration_slug_prefix = $registration_slug_prefix;
```

Comment:
please do a trim and UPPER CASE HERE also REPLACE spaces BY _

Reply:
done